util/Hrafnar/Util.hs

```diff
@@ -0,0 +1,26 @@
+module Hrafnar.Util
+  ( rawS
+  , rawC
+  ) where
+
+import           Language.Haskell.TH.Quote
+import qualified Language.Haskell.TH.Syntax as S
+
+genQQ :: S.Lift a => (String -> a) -> QuasiQuoter
+genQQ parseFunc =
+  QuasiQuoter
+    { quoteExp = S.lift . parseFunc
+    , quotePat = undefined
+    , quoteType = undefined
+    , quoteDec = undefined
+    }
+
+-- raw string
+rawS :: QuasiQuoter
+rawS = genQQ id
+
+-- raw char
+rawC :: QuasiQuoter
+rawC = genQQ head
+-- HACK: `head` does not raise an error when a given string is more than
+-- 1 charcter. It smells bad.
```

Comment:
Do you have any idea?

Reply:
Define `head` manually like this:
```haskell
head' :: [a] -> Maybe a
head' [] = Nothing
head' (x:_) = Just x
```
Or [this package](https://hackage.haskell.org/package/safe) suite it.

Reply:
Hmm... Isn't there any suitable function in built-in functions?
I think this conversion is a basic operation...

OK, I'll define it manually.

Reply:
I've implemented it in cff2054 .

cf. execution results in ghci

```shell
*Hrafnar.Util> [rawC||]

<interactive>:2:1: error:
    • Exception when trying to run compile-time code:
        no character
CallStack (from HasCallStack):
  error, called at /Users/ikngtty/Projects/src/github.com/realglobe-Inc/hrafnar/util/Hrafnar/Util.hs:27:17 in main:Hrafnar.Util
      Code: quoteExp rawC ""
    • In the quasi-quotation: [rawC||]
*Hrafnar.Util> [rawC|a|]
'a'
*Hrafnar.Util> [rawC|ab|]

<interactive>:4:1: error:
    • Exception when trying to run compile-time code:
        multiple characters
CallStack (from HasCallStack):
  error, called at /Users/ikngtty/Projects/src/github.com/realglobe-Inc/hrafnar/util/Hrafnar/Util.hs:29:20 in main:Hrafnar.Util
      Code: quoteExp rawC "ab"
    • In the quasi-quotation: [rawC|ab|]
```